src/network/ClientConnectionManager.ts

```diff
@@ -264,6 +264,12 @@ export class ClientConnectionManager extends EventEmitter {
             })
             .then((socket) => {
                 clientConnection = new ClientConnection(this.client, translatedAddress, socket, this.connectionIdCounter++);
+                // close the connection proactively on errors
+                socket.on('error', (err: any) => {
+                    if (err.code === 'EPIPE' || err.code === 'ECONNRESET') {
```

Comment:
I know it was like this before (I also didn't change this part), but should we close the connection here without checking the error code? The documentation of the `error` event says that `Emitted when an error occurs. The 'close' event will be called directly following this event.` So the question is after we receive an error event here(which is not `EPIPE` or 
 `ECONNRESET`, ), is it possible to continue operating?

Reply:
Yes, I had a similar question about that check, but decided to keep things as were. `git blame` tells me that it was like this since the initial error handling implementation and I failed to find a good reason for this check. So, I'm going to remove it. Thanks for bringing this question to the table.

Reply:
Addressed in 39c99306f6c2475ee53a95f4de9a06c1bcac2ce6